From d3712b6a613e2af059e947baf0c264d5b8eb5e99 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Mon, 20 May 2019 16:03:40 +0900
Subject: [PATCH 7/8] shared/sec: rename func for parsing MAC addr

This update is to rename `decode_mac_addr_str_value()` to
`parse_mac_addr()` and refactor the function.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 .../secondary/spp_worker_th/cmd_parser.c      | 21 ++++++++++---------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/src/shared/secondary/spp_worker_th/cmd_parser.c b/src/shared/secondary/spp_worker_th/cmd_parser.c
index fc30463..064b18f 100644
--- a/src/shared/secondary/spp_worker_th/cmd_parser.c
+++ b/src/shared/secondary/spp_worker_th/cmd_parser.c
@@ -597,22 +597,23 @@ parse_port_pcp(void *output, const char *arg_val,
 	return SPP_RET_OK;
 }
 
-/* decoding procedure of mac address string */
+/* Parse mac address string. */
 static int
-decode_mac_addr_str_value(void *output, const char *arg_val,
-				int allow_override __attribute__ ((unused)))
+parse_mac_addr(void *output, const char *arg_val,
+		int allow_override __attribute__ ((unused)))
 {
-	int64_t ret = SPP_RET_OK;
+	int64_t res;
 	const char *str_val = arg_val;
 
-	/* if default specification, convert to internal dummy address */
+	/* If given value is the default, use dummy address instead. */
 	if (unlikely(strcmp(str_val, SPP_DEFAULT_CLASSIFIED_SPEC_STR) == 0))
 		str_val = SPP_DEFAULT_CLASSIFIED_DMY_ADDR_STR;
 
-	ret = spp_change_mac_str_to_int64(str_val);
-	if (unlikely(ret < SPP_RET_OK)) {
+	/* Check if the given value is valid. */
+	res = spp_change_mac_str_to_int64(str_val);
+	if (unlikely(res < SPP_RET_OK)) {
 		RTE_LOG(ERR, SPP_COMMAND_PROC,
-				"Bad mac address string. val=%s\n", str_val);
+				"Invalid MAC address `%s`.\n", str_val);
 		return SPP_RET_NG;
 	}
 
@@ -759,7 +760,7 @@ cmd_ops_list[][SPPWK_MAX_PARAMS] = {
 			.name = "mac address",
 			.offset = offsetof(struct spp_command,
 					spec.cls_table.mac),
-			.func = decode_mac_addr_str_value
+			.func = parse_mac_addr
 		},
 		{
 			.name = "port",
@@ -792,7 +793,7 @@ cmd_ops_list[][SPPWK_MAX_PARAMS] = {
 			.name = "mac address",
 			.offset = offsetof(struct spp_command,
 					spec.cls_table.mac),
-			.func = decode_mac_addr_str_value
+			.func = parse_mac_addr
 		},
 		{
 			.name = "port",
-- 
2.17.1

